src/Umbraco.Core/DisposableObjectSlim.cs

```diff
@@ -0,0 +1,42 @@
+﻿using System;
+
+namespace Umbraco.Core
+{
+    public abstract class DisposableObjectSlim : IDisposable
+    {
+        private bool _disposed;
+        private readonly object _locko = new object();
+
+        // gets a value indicating whether this instance is disposed.
+        // for internal tests only (not thread safe)
+        //TODO make this internal + rename "Disposed" when we can break compatibility
+        public bool IsDisposed { get { return _disposed; } }
```

Comment:
Please follow the TODO instructions for this method :)

Reply:
Done